geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterStore.scala

```diff
@@ -0,0 +1,366 @@
+                                 timings.time("planning"),
+                                 timings.time("scanning") - timings.time("planning"),
+                                 timings.time("mosaic"),
+                                 numRasters)
+      this.writeStat(stat, profileTable)
+    }
+    image
+  }
+
+  def getRasters(rasterQuery: RasterQuery)(implicit timings: Timings): Iterator[Raster] = {
+    profile({
+      val batchScanner = connector.createBatchScanner(tableName, authorizationsProvider.getAuthorizations, numQThreads)
+      val plan = queryPlanner.getQueryPlan(rasterQuery, getAvailabilityMap)
+      plan match {
+        case Some(qp) =>
+          configureBatchScanner(batchScanner, qp)
+          adaptIteratorToChunks(SelfClosingBatchScanner(batchScanner))
+        case _        => Iterator.empty
+      }
+    }, "scanning")
+  }
+
+  def getQueryRecords(numRecords: Int): Iterator[String] = {
+    val scanner = connector.createScanner(profileTable, authorizationsProvider.getAuthorizations)
+    scanner.iterator.take(numRecords).map(RasterQueryStatTransform.decodeStat)
+  }
+
+  def getBounds: BoundingBox = {
+    ensureBoundsTableExists()
+    val scanner = connector.createScanner(GEOMESA_RASTER_BOUNDS_TABLE, authorizationsProvider.getAuthorizations)
+    scanner.setRange(new Range(getBoundsRowID))
+    val resultingBounds = SelfClosingScanner(scanner)
+    if (resultingBounds.isEmpty) {
+      BoundingBox(-180, 180, -90, 90)
+    } else {
+      //TODO: GEOMESA-646 anti-meridian questions
+      reduceValuesToBoundingBox(resultingBounds.map(_.getValue))
+    }
+  }
+
+  // TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
+  def getAvailableResolutions: Seq[Double] = getAvailabilityMap.keySet.toSeq.sorted
+
+  def getAvailableGeoHashLengths: Set[Int] = getAvailabilityMap.values().toSet
+
+  def getAvailabilityMap: ImmutableSetMultimap[Double, Int] = AccumuloRasterStore.boundsCache.get(tableName, callable)
+
+  def callable = new Callable[ImmutableSetMultimap[Double, Int]] {
+    override def call(): ImmutableSetMultimap[Double, Int] = {
+      ensureBoundsTableExists()
+      val scanner = connector.createScanner(GEOMESA_RASTER_BOUNDS_TABLE, getAuths)
+      scanner.setRange(new Range(getBoundsRowID))
+      val scanResultingKeys = SelfClosingScanner(scanner).map(_.getKey).toSeq
+      val geohashlens = scanResultingKeys.map(_.getColumnFamily.toString).map(lexiDecodeStringToInt)
+      val resolutions = scanResultingKeys.map(_.getColumnQualifier.toString).map(lexiDecodeStringToDouble)
+      val m = new ImmutableSetMultimap.Builder[Double, Int]()
+      (resolutions zip geohashlens).foreach(x => m.put(x._1, x._2))
+      m.build()
+    }
+  }
+
+  def getGridRange: GridEnvelope2D = {
+    val bounds = getBounds
+    val resolutions = getAvailableResolutions
+    // If no resolutions are available, then we have an empty table so assume 1.0 for now
```

Comment:
A default resolution parameter should likely be set elsewhere, like at the top of the class. If this assumption is also made elsewhere, then a linkage needs to be made somehow..